collector/selinux_linux.go

```diff
@@ -0,0 +1,76 @@
+// Copyright 2021 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build linux && !noselinux
+// +build linux,!noselinux
+
+package collector
+
+import (
+	"github.com/go-kit/log"
+	"github.com/opencontainers/selinux/go-selinux"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type selinuxCollector struct {
+	configMode  *prometheus.Desc
+	currentMode *prometheus.Desc
+	status      *prometheus.Desc
+	logger      log.Logger
+}
+
+func init() {
+	registerCollector("selinux", defaultEnabled, NewSelinuxCollector)
+}
+
+// NewSelinuxCollector returns a new Collector exposing SELinux statistics.
+func NewSelinuxCollector(logger log.Logger) (Collector, error) {
+	return &selinuxCollector{
+		configMode: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "selinux", "config_mode"),
+			"Configured SELinux enforcement mode",
+			nil, nil,
+		),
+		currentMode: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "selinux", "current_mode"),
+			"Current SELinux enforcement mode",
+			nil, nil,
+		),
+		status: prometheus.NewDesc(
+			prometheus.BuildFQName(namespace, "selinux", "status"),
+			"SELinux status",
```

Comment:
I think this would be more clear if it was called `enabled`, this is also what the library calls it.

```suggestion
			prometheus.BuildFQName(namespace, "selinux", "enabled"),
			"SELinux is enabled, 1 is true, 0 is false",
```